GUIDE.md

```diff
@@ -0,0 +1,1081 @@
+
+You can explicitly disable css-modules on an import by adding `?modules=false` to the import path. For example
+```javascript
+import style from './style.scss?modules=false';
+```
+
+Styling components is an area with a lot of differing opinions and available options in the community. Using the plugin as-is will **save you a lot of time** unless you strictly need something else.
+
+## Special variables
+
+Webpack is used for creating your application bundles and we make use of the [DefinePlugin](http://webpack.github.io/docs/list-of-plugins.html#defineplugin)
+
+The following useful constants are available to you in your project code:
+- \_\_WEB\_\_
+  - Boolean. `true` if client bundle.
+- \_\_NODE\_\_
+  - Boolean. `true` if server bundle
+
+These are extra useful because they are evaluated at `compile time`. They can be used to make sure code is only included in a specific bundle. For example:
+
+```javascript
+if (__WEB__) {
+  console.log('I will only exist in the client bundle.');
+}
+
+if (__NODE__) {
+  console.log('I will only exist in the server bundle.');
+}
+```
+
+Will in your final **server** bundle be:
+
+```javascript
+console.log('I will only exist in the server bundle.');
+```
+
+## Adding extra resources
+
+If you need to include extra resources like global styles or scripts into the start of your application bundle this can be done using the `--build-resources` configuration.
+
+```
+--build-resources    An array of files to include into the build process.
+```
+
+Note that the default value for this configuration is `["roc-package-web-app-react/styles/base.css"]` which contains some [css-resets](https://github.com/rocjs/roc-package-web-app-react/blob/master/extensions/roc-package-web-app-react/styles/base.css). If you wish to add more resources by redefining this array be sure to include this default explicitly, otherwise you will lose the resets. For example `["roc-package-web-app-react/styles/base.css", "styles/somethingelse.css"]`
+
+## Extending the root HTML document
+
+A complete server-rendered HTML document is provided by `web-app-react`. The ([njk](https://mozilla.github.io/nunjucks/)) template used to render this document [can be viewed on github](https://github.com/rocjs/roc-package-web-app-react/blob/master/extensions/roc-package-web-app-react/views/roc-package-web-app-react/main.njk).
+
+You can extend this in your application using several mechanisms:
+
+### Define individual custom template values
+```
+--build-templateValues
+```
+
+This defaults to ``"src/template-values.js"``. In this file you can define a function that returns an object with values that will be exposed in the template as `custom`.
+
+Example `src/template-values.js`:
+```javascript
+export default function getTemplateValues({ koaState, settings, reduxState }) {
+  // you can use state from koa, settings and redux state to make your decision
+  return {
+      bodyClass: 'main',
```

Comment:
This is a somewhat undocumented feature, should we perhaps use something else for the example?

Reply:
Sure. Suggestions on a suitable alternative?

Reply:
Adding a note that it will likely be removed in the future.